tests/audiohelper.js

```diff
@@ -512,6 +520,31 @@ describe('AudioHelper', () => {
       });
 
       describe('setInputDevice', () => {
+        it('should wait for the beforeSetInputDevice to resolve', async () => {
+          const stub = sinon.stub();
+          audio = new AudioHelper(onActiveOutputsChanged, onActiveInputChanged, {
+            audioProcessorEventObserver: eventObserver,
+            beforeSetInputDevice: () => new Promise(res => res(stub())),
+            getUserMedia,
+            mediaDevices,
+            setSinkId: () => {}
+          });
+          await audio._setInputDevice('input');
+          sinon.assert.calledOnce(stub);
+        });
+        
+        it('should reject if beforeSetInputDevice rejects', async () => {
+          const stub = sinon.stub();
+          audio = new AudioHelper(onActiveOutputsChanged, onActiveInputChanged, {
+            audioProcessorEventObserver: eventObserver,
+            beforeSetInputDevice: () => new Promise((resolve, reject) => reject()),
+            getUserMedia,
+            mediaDevices,
+            setSinkId: () => {}
+          });
+          await assert.rejects(() => audio._setInputDevice('input'));
```

Comment:
Let's use the public API
```suggestion
          await assert.rejects(() => audio.setInputDevice('input'));
```